Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,493 @@
+        self.tracker_interval = 200
+        self.logging_interval = 60
+        self.share_mode_target = 3
+        self.policy = policy(session)
+
+        # Non-Configurable
+        self.initial_logging_interval = 20
+        self.initial_tracker_interval = 25
+        self.initial_swarm_interval = 30
+        self.min_connection_start = 5
+        self.min_channels_start = 100
+        self.credit_mining_path = os.path.join(DefaultDownloadStartupConfig.getInstance().get_dest_dir(),
+                                               CREDIT_MINING_FOLDER_DOWNLOAD)
+        self.load_config = load_config
+
+
+class BoostingManager(TaskManager):
+    """
+    Class to manage all the credit mining activities
+    """
+
+    __single = None
+
+    def __init__(self, session, settings=None):
+        super(BoostingManager, self).__init__()
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        BoostingManager.__single = self
+        self.boosting_sources = {}
+        self.torrents = {}
+
+        self.session = session
+        assert self.session.get_libtorrent()
+
+        self.torrent_db = self.session.open_dbhandler(NTFY_TORRENTS)
+        self.channelcast_db = self.session.open_dbhandler(NTFY_CHANNELCAST)
+
+        # use provided settings or a default one
+        self.settings = settings or BoostingSettings(session, load_config=True)
+
+        if self.settings.load_config:
+            self._logger.info("Loading config file from session configuration")
+            self.load_config()
+
+        if not os.path.exists(self.settings.credit_mining_path):
+            os.makedirs(self.settings.credit_mining_path)
+
+        self.session.lm.ltmgr.get_session().set_settings(
+            {'share_mode_target': self.settings.share_mode_target})
+
+        self.session.add_observer(self.on_torrent_notify, NTFY_TORRENTS, [NTFY_UPDATE])
+
+        self.register_task("CreditMining_select", LoopingCall(self._select_torrent),
+                           self.settings.initial_swarm_interval, interval=self.settings.swarm_interval)
+
+        self.register_task("CreditMining_scrape", LoopingCall(self.scrape_trackers),
+                           self.settings.initial_tracker_interval, interval=self.settings.tracker_interval)
+
+        self.register_task("CreditMining_log", LoopingCall(self.log_statistics),
+                           self.settings.initial_logging_interval, interval=self.settings.logging_interval)
+
+    @staticmethod
+    def get_instance(*args, **kw):
+        """
+        get single instance of Boostingmanager
+        """
+        if BoostingManager.__single is None:
+            BoostingManager(*args, **kw)
+        return BoostingManager.__single
+
+    @staticmethod
+    def del_instance():
+        """
+        resetting, then deleting single instance
+        """
+        BoostingManager.__single = None
+
+    def shutdown(self):
+        """
+        save configuration before stopping stuffs
```

Comment:
stuffs sounds funny but I think it should be something like "before stopping the boostingmanager"  or something like that